corehq/apps/geospatial/static/geospatial/js/gps_capture.js

```diff
@@ -101,27 +101,25 @@ hqDefine("geospatial/js/gps_capture",[
             self.goToPage(1);
         };
 
-        self.onValueChanged = function () {
-            self.hasUnsavedChanges(true);
-        };
-
-        self.onSaveClicked = function () {
+        self.saveDataRow = function (dataItem) {
+            self.isSubmissionSuccess(false);
             self.hasSubmissionError(false);
             $.ajax({
                 method: 'POST',
                 url: initialPageData.reverse('gps_capture'),
                 data: JSON.stringify({
                     'data_type': self.dataType,
-                    'data_items': ko.mapping.toJS(self.dataItems()),
+                    'data_item': ko.mapping.toJS(dataItem),
                 }),
                 dataType: "json",
                 contentType: "application/json; charset=utf-8",
                 success: function () {
-                    location.reload();
+                    dataItem.hasUnsavedChanges(false);
+                    self.isSubmissionSuccess(true);
                 },
                 error: function (e) {
+                    console.error("Error in submission", e);
```

Comment:
This was left intentionally as I thought it could be useful for quick debugging from the browser console if anything ever does go wrong. I'm happy to remove this however, if it's better practice to not have console logs.

Reply:
> if it's better practice to not have console logs

Yup.

Reply:
Thanks for the clarification, they have been removed ([faf5ba8](https://github.com/dimagi/commcare-hq/pull/33420/commits/faf5ba8ee7afbc653e87e6c7424426c50a071179)).